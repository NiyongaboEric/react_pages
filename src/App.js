import React from 'react'
import Pagination from './pagination'

const App = () => (
  <Pagination
    itemsPerPage={1}
    activeStyle={{ backgroundColor: '#00b9f2', color: 'white' }}
    next="next"
    prev="previous"
    data={[
      { firstName: 'John', lastName: 'Doe' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
      { firstName: 'Nhoj', lastName: 'Eod' },
    ]}
    oneItem={item => (
      <div>
        <div>{item.firstName}</div>
        <div>{item.lastName}</div>
      </div>
    )}
  />
)

export default App
