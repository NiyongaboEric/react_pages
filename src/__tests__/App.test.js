import React from 'react'
import { shallow, mount } from 'enzyme'
import App from '../App'

describe('App test.', () => {
  test('Buttons and snapshot when props is provided', () => {
    const wrapper = mount(
      <App
        itemsPerPage={1}
        activePageStyle={{ backgroundColor: '#00b9f2', color: 'white' }}
        next="next"
        prev="previous"
        data={[{ firstName: 'John', lastName: 'Doe' },{ firstName: 'Alice', lastName: 'Bot' }]}
        pageButtons={7}
        onePage={(item, index) => (
          <div key={index}>
            <div>{item.firstName}</div>
            <div>{item.lastName}</div>
          </div>
        )}
      />
    )
    const page1 = wrapper.find('button').at(1)
    const page2 = wrapper.find('#prev')
    const page3 = wrapper.find('#next')
    page1.simulate('click')
    page2.simulate('click')
    page3.simulate('click')
    expect(jest.fn().mock.calls).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  test('Test snapshot when props is not provided', () => {
    const wrapper = mount(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
