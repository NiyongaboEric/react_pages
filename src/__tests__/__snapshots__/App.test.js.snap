// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test. Buttons and snapshot when props is provided 1`] = `ReactWrapper {}`;

exports[`App test. Test snapshot when props is not provided 1`] = `ReactWrapper {}`;
